test/unit/org/apache/cassandra/db/VerifyTest.java

```diff
@@ -758,6 +781,65 @@ public void testNoFilterFile()
         }
     }
 
+    @Test
+    public void testVerifyWithoutRealm()
+    {
+        CompactionManager.instance.disableAutoCompaction();
+        Keyspace keyspace = Keyspace.open(KEYSPACE);
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF);
+
+        fillCF(cfs, 2);
+
+        Descriptor descriptor = cfs.getLiveSSTables().iterator().next().getDescriptor();
+        SSTableReader sstable = descriptor.getFormat()
+                                          .getReaderFactory()
+                                          .openNoValidation(descriptor,
+                                                            SSTableReader.componentsFor(descriptor),
+                                                            TableMetadataRef.forOfflineTools(cfs.metadata()));
+
+        try (Verifier verifier = new Verifier(sstable,
+                                              true,
+                                              Verifier.options().invokeDiskFailurePolicy(true).build()))
+        {
+            verifier.verify();
+        }
+        catch (CorruptSSTableException err)
+        {
+            fail("Unexpected CorruptSSTableException");
+        }
+    }
+
+    @Test
+    public void testVerifyWithoutRealmMismatch()
+    {
+        System.setProperty(SSTableFormat.FORMAT_DEFAULT_PROP, SSTableFormat.Type.BIG.name);
+        CompactionManager.instance.disableAutoCompaction();
+        Keyspace keyspace = Keyspace.open(KEYSPACE);
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF);
+
+        fillCF(cfs, 2);
+
+        Collection<SSTableReader> sstables = SSTableReader.selectOnlyBigTableReaders(cfs.getLiveSSTables(), Collectors.toList());
+        assertEquals(1, sstables.size());
+        Descriptor descriptor = sstables.iterator().next().getDescriptor();
+        SSTableReader sstable = descriptor.getFormat()
+                                          .getReaderFactory()
+                                          .openNoValidation(descriptor,
+                                                            SSTableReader.componentsFor(descriptor),
+                                                            TableMetadataRef.forOfflineTools(cfs.metadata()));
+
+        try (Verifier verifier = new Verifier(sstable,
+                                              new OutputHandler.CustomLogOutput(LoggerFactory.getLogger(Verifier.class)),
+                                              true,
+                                              Verifier.options().invokeDiskFailurePolicy(true).build()))
+        {
+            verifier.verify();
+        }
+        catch (CorruptSSTableException err)
+        {
+            fail("Unexpected CorruptSSTableException");
+        }
+    }
```

Comment:
The DSE version also added this to `testVerifyMutateRepairStatus` - was it overlooked or dropped intentionally?

```

        // Check that is not possible to create a Verifier without passing a ColumnFamilyStore if mutateRepairStatus is true
        assertThrows(IllegalArgumentException.class, () -> new Verifier(sstable, false, Verifier.options().mutateRepairStatus(true).build()));
```

Reply:
Yes, I missed it.

Reply:
I added a separate test, so it is more clear about its semantics.